### ISSUE TITLE
waitSignal cluster.yaml section is not - respected in the template

### ISSUE THREAD

**Issue by javapapo** [coreos/kube-aws#371]
Assuming that you have the following section configured on your cluster.yaml 
```
# When enabled, autoscaling groups managing controller nodes wait for nodes to be up and running.
# It is enabled by default.
waitSignal:
  enabled: false
  maxBatchSize: 1
```
Unfortunately on the generated cloudformation (after the stack is rendered) - or even in the intermediate template. We always have the section generated! 

For example looking at the sections of `node-pool.json.tmpl`

```
{{if .WaitSignal.Enabled}}
          "WaitOnResourceSignals" : "true",
          "MaxBatchSize" : "{{.WaitSignal.MaxBatchSize}}",
          "PauseTime": "{{.CreateTimeout}}"
          {{else}}
```
it seems that this part is always rendered on the generate stack.json.

Since this, part is causing us problems on our AWS account/setup, we need to manually remove the above part from the `node-pool.json.tmpl` so that is not rendered at all. 

Thanks for your time anyway & and many many thanks for your great tool!

**Comment by Camsteack** [comment]
Hi @gianrubio

This is not really what we need. The issue is that the Waitsignal is not working and is causing the stack to fail as it does not get a valid response from all the worker node, even if they are properly created.
We want to be able to disable it using
```
waitSignal:
  enabled: false
```

But it is not working so we have to manually remove it from the stack template.
Thanks a lot.

**Comment by gianrubio** [comment]
@Camsteack I accidentally pushed the code, I haven't finished yet. Sorry

**Comment by mumoshu** [comment]
Hi! Just a quick reply but as the comment implies, please try changing
`worker.nodePools[].waitSignal` to disable it for a node pool.
2017年3月2日(木) 0:22 Camille <notifications@github.com>:

**Comment by Camsteack** [comment]
We will try that. Thanks a lot @mumoshu and @gianrubio
We are having the same issue with :
```
clusterAutoscalerSupport:
        enable: true
```

I don't know if we are doing something wrong in the syntax but it is not picked up but the stack template either.

**Comment by mumoshu** [comment]
@javapapo @Camsteack Did it work for you? Basically, you have to differentiate the top level `experimental.waitSignal` and the node-pool specific `worker.nodePools[].waitSignal` according to their comments inside cluster.yaml.

**Comment by javapapo** [comment]
No, unfortunately

**Comment by mumoshu** [comment]
Thanks for the confirmation @javapapo!
Now, adding more tests revealed a bug which had been making waitSignal unable to be disabled.
The bug is fixed in #371

**Comment by mumoshu** [comment]
@javapapo The fix is included in v0.9.5-rc.2.
Sorry for taking your time so much but could you please confirm if it works for you?

**Comment by javapapo** [comment]
Thank you for your hard work and the replies! Will try it asap

**Comment by javapapo** [comment]
So on the root level of my cluster yaml I have
`
waitSignal:
  enabled: false
  maxBatchSize: 1
`

On the 
`  
"Resources": {
    "Controllers": {
      "Type": "AWS::AutoScaling::AutoScalingGroup",
`

I see
`
"UpdatePolicy": {
        "AutoScalingRollingUpdate": {
          "MinInstancesInService": "2",
          "MaxBatchSize": "1",
          "WaitOnResourceSignals": "true",
          "PauseTime": "PT15M"
        }
`

Or Should I just place it on ?

`controller:
     autoScalingGroup:
    waitSignal:
      enabled: false
     maxBatchSize: 1
`

**Comment by javapapo** [comment]
Apart from the above, the cluster creation was successful, so I think the issue is resolved

**Comment by mumoshu** [comment]
@javapapo 

> So on the root level of my cluster yaml I have
> ```yaml
> waitSignal:
>   enabled: false
>   maxBatchSize: 1
> ```

Yes, this is correct.
And sorry but I've just noticed that I've mistakenly missed the commit to fix this in the v0.9.5-rc.2 release!
v0.9.5-rc.3 released today does include the fix
